torchdata/dataloader2/reading_service.py

```diff
@@ -306,6 +308,7 @@ def initialize_iteration(self) -> None:
         shared_seed_int: int = shared_seed.item()  # type: ignore[assignment]
         _seed_generator = torch.Generator()
         _seed_generator.manual_seed(shared_seed_int)
+        self._initial_seed = shared_seed_int
```

Comment:
The way I should get original state, prior to `generate_random_scalar_tensor` I should get PyTorch global random state. 

When I restore, my restoration function should restore that state prior to calling `initialize_iteration`

Reply:
Think about what user will call (probably `iter(dl2)`), and how that will trigger various initialization methods